### PR TITLE
Add licenses.db; fix packaging; recognize more file patterns

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- Fixes https://github.com/conda-forge/staged-recipes/pull/12558#issuecomment-693380650 :
  This now includes the `licenses.db` which wasn't added to the Conda package.

- Addresses https://github.com/conda-forge/staged-recipes/pull/12558#issuecomment-693155109 :
  We shouldn't write to the recipe folder during build. Including files via the recipe folder works, but that's an implicit behavior and not an explicit one we should depend on (`conda-build` could be changed to, e.g., package these files before the actual build; and if someone built with `conda-build --no-include-recipe` the licenses wouldn't be included.).
  The way it currently is, is not optimal, but I opened https://github.com/conda/conda-build/issues/4121 to track this use case.
  I'm using the `gather_licenses` function included in this PR in other recipes that I'm about to submit to staged-recipes.

### CODE REVIEW

**Comment by mbargull** [comment]
Windows build fails because of https://github.com/google/licenseclassifier/blob/842c0d70d7027215932deb13801890992c9ba364/file_system_resources.go#L49 which assumes absolutes path to start with `/` ... which of course does not recognize the `D:/...` on Windows...

File: recipe/build.sh [hunk 1]
```diff
@@ -5,11 +5,16 @@ set -ex
 GOBIN=$(go env GOBIN)
 export GOBIN=$GOBIN
 
+go get -d github.com/google/go-licenses
+# Copy our patched vendored source before starting the build.
+cp -r \
+  licenseclassifier-${licenseclassifier_commit}/* \
+  "${GOPATH:-"$( go env GOPATH )"}"/pkg/mod/github.com/google/licenseclassifier@*-${licenseclassifier_commit}/
```
Comment (mbargull) [inline recipe/build.sh]: This feels rather awkward... Well, I'm no Go dev and can't tell what other solution would look better :shrug:.

Comment (mbargull) [reply recipe/build.sh]: I intend to upstream this and also the path for `go-licenses` -- but that involves signing a CLA for Google and as such can take a bit of time.
Let's go with the patches for now and clean up afterwards.

**Comment by mbargull** [comment]
I'm going ahead with the merge (to put this one to use for the https://github.com/containers packages I've worked on).
I'll clean up the patching once everything is upstreamed.

**Comment by dhirschfeld** [comment]
Thanks for picking this up @mbargull! I got it to a point but found it didn't actually work in practice and as I have literally zero `go` knowledge I didn't know how to fix it myself.

I think it will be very useful for `go` recipes so I'm glad you took up the challenge and sorted it out! :+1:

**Comment by mbargull** [comment]
> I have literally zero `go` knowledge

Same here :D.
I guess we just have to add it to the builds of some feedstocks and hope an eager Go developer then notices it and adds themselves as a maintainer here to improve things if I messed something up ;P.

**Pull request merged**